lesson_41/second-react-app/src/App.jsx

```diff
@@ -1,36 +1,36 @@
 import React from "react"
-import {useState, useEffect} from "react"
+import {useEffect} from "react"
 import Header from "./components/header/Header.jsx"
 import Products from "./components/products/Products.jsx"
 import CreateProduct from "./components/createProducts/CreateProduct.jsx"
+import {useSelector, useDispatch} from "react-redux"
+import {increment, addProduct} from "./redux/cart"
 import "./App.css"
 
-export const addNewProduct = React.createContext()
-
 export const App = () => {
-  const [productsIn, setProducts] = useState([])
-  const [productsInCart, setProductsInCart] = useState([])
+  const productsIn = useSelector((state) => state.cart.products)
+  const dispatch = useDispatch()
 
   useEffect(() => {
     fetch("https://dummyjson.com/products")
       .then((response) => response.json())
       .then((data) => {
         let fetchProducts = data.products
-        setProducts(fetchProducts)
+        fetchProducts.forEach((product) => {
+          dispatch(addProduct(product))
```

Comment:
better to write a function which set all products by one dispatch call

what if you will have 10 000 products? will you trigger that dispatch 10 000 times?:)